src/broadcast/broadcast.rs

```diff
@@ -26,16 +26,30 @@ pub struct Broadcast<N> {
     proposer_id: N,
     /// The Reed-Solomon erasure coding configuration.
     coding: Coding,
+    /// Original value message received from proposer.
+    value_message: Proof<Vec<u8>>,
     /// If we are the proposer: whether we have already sent the `Value` messages with the shards.
     value_sent: bool,
-    /// Whether we have already multicast `Echo`.
-    echo_sent: bool,
+    /// Whether we have already multicast `Echo` to our left nodes.
+    echo_sent_partial: bool,
+    /// Whether we have already multicast `Echo` to our right nodes who haven't sent `CanDecode`.
+    echo_sent_full: bool,
     /// Whether we have already multicast `Ready`.
     ready_sent: bool,
+    /// Whether we have already multicast `EchoHash` to the right nodes.
+    echo_hash_sent: bool,
```

Comment:
That should always be identical to `echo_sent_partial`, I think?

Reply:
Yes you are right. It also serves no purpose in the rest of the code as of now.